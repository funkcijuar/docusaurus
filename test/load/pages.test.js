import loadPages from '@lib/load/pages';
import path from 'path';

describe('loadPages', () => {
  test('valid pages', async () => {
    const pagesDir = path.join(__dirname, '__fixtures__', 'simple-pages');
    const pagesData = await loadPages(pagesDir);
    pagesData.sort((a, b) => a.path > b.path); // because it was unordered
    expect(pagesData).toEqual([
      {
        path: '/',
        source: 'index.js'
      },
      {
        path: '/bar/baz',
        source: 'bar/baz.js'
      },
      {
        path: '/foo',
        source: 'foo.js'
      },
      {
        path: '/foo/',
        source: 'foo/index.js'
      }
    ]);
    expect(pagesData).not.toBeNull();
  });

  test('invalid pages', async () => {
    const nonExistingDir = path.join(__dirname, '__fixtures__', 'nonExisting');
    const pagesData = await loadPages(nonExistingDir);
    expect(pagesData).toEqual([]);
  });
});
