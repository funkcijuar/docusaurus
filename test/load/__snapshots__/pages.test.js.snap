// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loadPages valid pages 1`] = `
Array [
  Object {
    "path": "/foo",
    "source": "foo.js",
  },
  Object {
    "path": "/",
    "source": "index.js",
  },
  Object {
    "path": "/foo/",
    "source": "foo/index.js",
  },
  Object {
    "path": "/bar/baz",
    "source": "bar/baz.js",
  },
]
`;
